core/renderer/backend/opengl/RenderTargetGL.cpp

```diff
@@ -44,40 +44,45 @@ void RenderTargetGL::unbindFrameBuffer() const
 }
 
 void RenderTargetGL::update() const {
-    if (!_dirty) return;
-    if(!_defaultRenderTarget) {
-        { // color attachments
+    if (!_dirtyFlags) return;
+    if (!_defaultRenderTarget)
+    {
+        {  // color attachments
             GLenum bufs[MAX_COLOR_ATTCHMENT] = {GL_NONE};
             for (size_t i = 0; i < MAX_COLOR_ATTCHMENT; ++i)
             {
-                GLuint texture = 0;
-                GLint level = 0;
-                if (_color[i])
-                {
-                    texture = static_cast<GLuint>(_color[i].texture->getHandler());
-                    level = _color[i].level;
-                    bufs[i] = GL_COLOR_ATTACHMENT0 + i;
-                }
-                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, texture, level);
+                if (!bitmask::any(_dirtyFlags, getMRTColorFlag(i)))
+                    continue;
+                auto textureInfo = _color[i];
+                auto textureHandler =
+                    static_cast<GLuint>(textureInfo.texture != nullptr ? textureInfo.texture->getHandler() : 0);
+                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, textureHandler,
+                                       textureInfo.level);
+                bufs[i] = GL_COLOR_ATTACHMENT0 + i;
```

Comment:
`bufs[i]` must be set if there's an attachment, even if it's not dirty. I think it should look like:

```cpp
                auto textureInfo = _color[i];
                if (textureInfo.texture != nullptr)
                    bufs[i] = GL_COLOR_ATTACHMENT0 + i;
                if (!bitmask::any(_dirtyFlags, getMRTColorFlag(i)))
                    continue;
                auto textureHandler =
                    static_cast<GLuint>(textureInfo.texture != nullptr ? textureInfo.texture->getHandler() : 0);
                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, textureHandler,
                                       textureInfo.level);
```

Reply:
Do you mean case if just depth/stencil was changed? May be it is better to do _bufs_ as member of RenderTarget class and clear/set corresponding constants according to setColorAttachment?